src/Message/CIMCreateCardRequest.php

```diff
@@ -344,4 +328,69 @@ public function makeUpdatePaymentProfileRequest($parameters)
         $obj->initialize(array_replace($this->getParameters(), $parameters));
         return $obj->send();
     }
+
+    /**
+     * Handle requests when the customer profile already exists.
+     *
+     * @return CIMCreatePaymentProfileResponse|CIMGetPaymentProfileResponse
+     */
+    private function handleExistingCustomerProfile()
+    {
+        $parameters = $this->getParameters();
+        $getProfileResponse = $this->makeGetProfileRequest($parameters);
+        $profileData = $getProfileResponse->getData();
+        $paymentProfiles = $profileData["profile"]["paymentProfiles"];
+
+        // If the max number of profiles is reached, attempt to update the matching profile if it exists.
+        if ($this->last4 && count($paymentProfiles) >= self::MAX_ALLOWED_PAYMENT_PROFILES) {
+            $customerPaymentProfileId = $getProfileResponse->getMatchingPaymentProfileId($this->last4);
```

Comment:
This may be problematic because whenever a CC is renewed after expiry, the last4 still stay the same but the expiry changes. So you'll need to compare the expiry as well.

Reply:
Essentially, even if the last4 are the same, it may necessitate a payment profile update if the expiry has changed.

Reply:
We were matching only with last4 in v1, doing the same in v2 as well.